src/main/java/org/killbill/billing/plugin/adyen/api/mapping/KlarnaPaymentMappingService.java

```diff
@@ -65,6 +77,46 @@ public static PaymentInfo toPaymentInfo(final String merchantAccount, final Stri
         return lineItems;
     }
 
+    private static List<LineItem> getLineAdjustments(final List<LineItem> lineItems) {
+        final Map<String, LineAdjustment> adjustmentMap = new HashMap<>();
+        final List<LineItem> adjustmentLines = new ArrayList<>();
+        if(lineItems!= null && lineItems.size() > 0) {
+            for(LineItem lineItem: lineItems) {
+                List<LineAdjustment> adjustments = lineItem.getAdjustments();
+                if(adjustments != null && adjustments.size() > 0) {
+                    for(LineAdjustment adjustment: adjustments) {
+                        if(adjustmentMap.containsKey(adjustment.getType())) {
+                            //update the amount for existing adjustment record
+                            LineAdjustment existingRecord = adjustmentMap.get(adjustment.getType());
+                            Long newAmount = adjustment.getAmount() + existingRecord.getAmount();
+                            existingRecord.setAmount(newAmount);
+                        } else {
+                            //add new adjustment record in map
+                            LineAdjustment newRecord = new LineAdjustment(adjustment.getAmount(),
+                                                                          adjustment.getType(),
+                                                                          adjustment.getDescription());
+                            adjustmentMap.put(adjustment.getType(), newRecord);
+                        }
+                    }
+                }
+            }
+
+            // add the adjustments lines as line items
+            final Collection<LineAdjustment> finalAdjustments = adjustmentMap.values();
+            for(LineAdjustment adjustment: finalAdjustments) {
+                final LineItem item = new LineItem();
+                item.setQuantity(1L);
+                item.setInventoryType("adjustment");
+                item.setId(adjustment.getType());
+                item.setAmountIncludingTax(adjustment.getAmount());
+                item.setDescription(adjustment.getDescription());
+                adjustmentLines.add(item);
+            }
+        }
+
+        return adjustmentLines.size() > 0 ? adjustmentLines : null;
```

Comment:
Instead of returning `null`, can we use `Optional` here?

Reply:
No need of `Optional` and `null`. You can return empty list in my opinion.
`Nulls` are always bad as return value and optional looks like an over kill for empty collection check.

Reply:
Returning empty.